lib/capistrano/recipes/deploy/strategy/jenkins_artifact.rb

```diff
@@ -4,23 +4,34 @@
 require 'jenkins_api_client'
 
 class ::JenkinsApi::Client::Job
+  def get_last_successful_build(job_name)
+    @logger.info "Obtaining last successful build number of #{job_name}"
+    @client.api_get_request("/job/#{path_encode(job_name)}/lastSuccessfulBuild")
+  end
+
   def get_last_successful_build_number(job_name)
     @logger.info "Obtaining last successful build number of #{job_name}"
     res = @client.api_get_request("/job/#{path_encode(job_name)}/lastSuccessfulBuild")
     res['number']
   end
 
   def find_artifact_with_path(job_name, relative_path)
-    current_build_number  = get_current_build_number(job_name)
-    job_path              = "job/#{path_encode job_name}/"
-    response_json         = @client.api_get_request("/#{job_path}#{current_build_number}")
+    response_json = get_last_successful_build(job_name)
     if response_json['artifacts'].none? {|a| a['relativePath'] == relative_path }
       abort "Specified artifact not found in curent_build !!"
     end
     jenkins_path          = response_json['url']
     artifact_path         = URI.escape("#{jenkins_path}artifact/#{relative_path}")
     return artifact_path
   end
+
+  def find_last_successful_artifact(job_name)
```

Comment:
oops, this method should be called instead of `find_artifact`:

https://github.com/aereal/capistrano-strategy-jenkins_artifact/pull/12/files#diff-36f6db7fc9bfde7999c0b847d9e222f2R74